src/Bicep.Cli/Services/Evaluation.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Deployments.Core.Definitions.Schema;
+
+namespace Bicep.Cli.Services
+{
+    public class Evaluation
+    {
+        public Template? Template { get; }
+        public InvalidOperationException? Error { get; }
```

Comment:
It's uncommon to store an exception rather than throwing it. Do you actually need this to be an exception, or could it just be a string?